src/config.c

```diff
@@ -1147,3 +1324,351 @@ RSTimeoutPolicy TimeoutPolicy_Parse(const char *s, size_t n) {
 void iteratorsConfig_init(IteratorsConfig *config) {
   *config = RSGlobalConfig.iteratorsConfigParams;
 }
+
+
+int RegisterModuleConfig(RedisModuleCtx *ctx) {
```

Comment:
Suggestion:
Have structs to represent config information for numeric, boolean, string and so on and have array for each type.
Then you can have a loop to register each config option instead of the blob of RM_TRY calls.

Its optional, you can resolve this thread if you feel it is too much work.